frame-metadata/src/v16.rs

```diff
@@ -186,7 +186,9 @@ impl IntoPortable for RuntimeApiMethodParamMetadata {
 )]
 pub struct ExtrinsicMetadata<T: Form = MetaForm> {
 	/// Extrinsic versions.
-	pub versions: Vec<u8>,
+	///
+	/// For each supported version number, list the indexes, in order, of the extensions used.
+	pub versions: BTreeMap<u8, Vec<u32>>,
```

Comment:
Just for clarity since it's in this struct along with non tx extension things, how about calling it `transaction_extensions_by_version` or something like that, and putting it right next to `transaction_extensions`?